CMakeLists.txt

```diff
@@ -118,12 +117,27 @@ if (HDF5_ENABLE_THREADSAFE AND NOT Threads_FOUND)
   find_package(Threads)
 endif ()
 
+set (CURL_TARGET_LIBTYPE shared)
+
+
 find_package (CURL 7.61 REQUIRED)
 if (CURL_FOUND)
+  # FindCURL module is not always found on windows; set paths manually
+  if (WIN32)
+    set (CURL_INCLUDE_DIRS "${CURL_ROOT}\\include")
+    set (LINK_LIBS ${LINK_LIBS} Ws2_32.lib Wldap32.lib)
```

Comment:
We shouldn't ever manually work around problems with a "Find" module like this. The issue is likely that cmake on those runners isn't new enough. Since FindCURL was introduced in version 3.12 of cmake, we should make that the minimum version for `cmake_minimum_required`.